zap/src/main/java/org/parosproxy/paros/extension/history/ExtensionHistory.java

```diff
@@ -881,13 +885,43 @@ public void eventReceived(Event event) {
      * @since 2.7.0
      */
     public void purgeHistory(List<HistoryReference> hrefs) {
-        if (hasView() && hrefs.size() > 1) {
-            int result =
-                    getView()
-                            .showConfirmDialog(
-                                    Constant.messages.getString("history.purge.warning"));
-            if (result != JOptionPane.YES_OPTION) {
-                return;
+        if (hrefs.isEmpty()) {
+            return;
+        }
+        if (hasView()) {
+            FileConfiguration config = Model.getSingleton().getOptionsParam().getConfig();
+            boolean confirmRemoval = config.getBoolean(REMOVE_CONFIRMATION_KEY, false);
```

Comment:
I think this was meant to be true by default, otherwise there's no prompt ever. (Same in the other dialogues.)